component/site/libraries/helper.php

```diff
@@ -487,18 +487,21 @@ function checkRobotsMetaTag($name = "robots", $content = "index,follow")
 	static public
 			function SetMetaTags()
 	{
+		// Get Global Config
+		$jConfig = JFactory::getConfig();
+
 		//Get Document to set the Meta Tags to.
 		$document = JFactory::getDocument();
 
 		//Get the Params.
 		$params = JComponentHelper::getParams(JEV_COM_COMPONENT);
 
-		if ($params->get('menu-meta_description') && !$document->getDescription())
+		if ($params->get('menu-meta_description') && $jConfig->get('MetaDesc', '') === $document->getDescription())
```

Comment:
Can I check (without running it through the debugger etc.)?

What you are testing is if the meta tags have been set to the same as the global default then we overwrite them with the menu specific values??  What if they are blank?

Reply:
@GeraintEdwards  The logic is that if there is no menu item meta and there is a global meta and the global matches what is set we kip.

If the global is blank then the document should be blank too, noting this is only setting if we have menu item set. We are just double checking the meta matches the global before setting it from the menu item.